File: MyMovies-MVP/app/src/main/java/com/jbrunton/mymovies/Movie.java
package com.jbrunton.mymovies;

import org.joda.time.DateTime;
import org.joda.time.LocalDate;

import java.util.Optional;

public class Movie {
    private final String id;
    private final String title;
    private final String posterPath;
    private final String backdropPath;
    private final LocalDate releaseDate;
    private final String rating;
    private final String overview;

    public Movie(String id, String title, String posterPath, String backdropPath, LocalDate releaseDate, String rating, String overview) {
        this.id = id;
        this.title = title;
        this.posterPath = posterPath;
        this.backdropPath = backdropPath;
        this.releaseDate = releaseDate;
        this.rating = rating;
        this.overview = overview;
    }

    public String getId() { return id; }

    public String getTitle() {
        return title;
    }

    public String getPosterPath() {
        return posterPath;
    }

    public String getBackdropPath() {
        return backdropPath;
    }

    public Optional<LocalDate> getReleaseDate() {
        return Optional.ofNullable(releaseDate);
    }

    public String getRating() {
        return rating;
    }

    public Optional<String> getOverview() {
        return Optional.ofNullable(overview);
    }
}

File: entities/src/main/java/com/jbrunton/entities/models/Movie.kt
package com.jbrunton.entities.models

import com.google.common.base.Optional
import org.joda.time.LocalDate

data class Movie(
    val id: String,
    val title: String,
    val posterUrl: Optional<String>,
    val backdropUrl: Optional<String>,
    val releaseDate: Optional<LocalDate>,
    val rating: String,
    val overview: Optional<String>) {
    companion object {
        val emptyMovie = Movie(
                id = "",
                title = "",
                posterUrl = Optional.absent(),
                backdropUrl = Optional.absent(),
                releaseDate = Optional.absent(),
                rating = "",
                overview = Optional.absent()
        )
    }
}
